OSSM-9066: Istioctl install instructions reference old Istio versions (#92133)
openshift/openshift-docs: modules/ossm-installing-the-istioctl-tool.adoc

Author: Now it appears.

// Module included in the following assemblies:
//
// * service-mesh-docs-main/install/ossm-istioctl-tool.adoc

:_mod-docs-content-type: PROCEDURE
[id="ossm-installing-the-istioctl-tool_{context}"]
= Installing the Istioctl tool

> Install the `istioctl` configuration command line utility to debug and diagnose {istio} service mesh deployments.

Reviewer: ```suggestion
Install the `istioctl` command-line utility to debug and diagnose {istio} service mesh deployments.
```

Author: done

.Prerequisites

* You have access to the {ocp-product-title} web console.

* The {SMProduct} 3 Operator is installed and running.

* You have created at least one `Istio` resource.

.Procedure

. Confirm which version of the `Istio` resource runs on the installation by running the following command:
+
[source,terminal]
----
$ oc get istio -ojsonpath="{range .items[*]}{.spec.version}{'\n'}{end}" | sed s/^v// | sort
----
+
If there are multiple `Istio` resources with different versions, choose the latest version. The latest version is displayed last.

> . In the {ocp-product-title} web console, click the Help icon and select *Command Line Tools*.

Reviewer: ```suggestion
. In the {ocp-product-title} web console, click the *Help* icon and select *Command Line Tools*.
```
The "Help" icon should be bold, right?

Author: Done

. Click *Download istioctl*. Choose the version and architecture that matches your system.
  
* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-linux-amd64.tar.gz[Linux (x86_64, amd64)]

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-linux-arm64.tar.gz[Linux on ARM (aarch64, arm64)]

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-darwin-amd64.tar.gz[MacOS (x86_64, amd64)]

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-darwin-arm64.tar.gz[MacOS on ARM (aarch64, arm64)]

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-windows-amd64.zip[Windows (x86_64, amd64)]

. Extract the `istioctl` binary file.

> .. If you are using a Linux Operating System, run the following command:

Reviewer: ```suggestion
.. If you are using a Linux operating system, run the following command:
```

See https://redhat-documentation.github.io/supplementary-style-guide/#operating-system

+
[source,terminal]
----
$ tar xzf istioctl-<VERSION>-<OS>-<ARCH>.tar.gz
----

> .. If you are using an Apple Mac Operating System, unpack and extract the archive.
> 
> .. If you are using a Microsoft Windows Operating System, use the zip software to extract the archive.

Reviewer: ```suggestion
.. If you are using an Apple Mac operating system, unpack and extract the archive.

.. If you are using a Microsoft Windows operating system, use the zip software to extract the archive.
```

See https://redhat-documentation.github.io/supplementary-style-guide/#operating-system

[id="ossm-installing-the-istioctl-tool_{context}"]
= Installing the Istioctl tool

Install the `istioctl` configuration command line utility to debug and diagnose {istio} service mesh deployments.

.Prerequisites

* You have access to the {ocp-product-title} web console.

* The {SMProduct} 3 Operator is installed and running.

* You have created at least one `Istio` resource.

.Procedure

. Confirm which version of the `Istio` resource runs on the installation by running the following command:
+
[source,terminal]
----
$ oc get istio -ojsonpath="{range .items[*]}{.spec.version}{'\n'}{end}" | sed s/^v// | sort
----
+
If there are multiple `Istio` resources with different versions, choose the latest version. The latest version is displayed last.

. In the {ocp-product-title} web console, click the Help icon and select *Command Line Tools*.

. Click *Download istioctl*. Choose the version and architecture that matches your system.
  

> * link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-linux-amd64.tar.gz[Linux (x86_64, amd64)]

Reviewer 2: Would it be possible to use `{istio-latest}` attribute for having dynamic version URLs? 

```suggestion
* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-{istio-latest}-linux-amd64.tar.gz[Linux (x86_64, amd64)]
```

Author: I made this change and created a preview build. Unfortunately, the logic in the tool chain does not appear to be able to resolve this and it results in a broken link.

Reviewer 2: That's a shame, it means we may have to update these links every new releases... For OSSM 3.0.2, it would already use `1.24.5` version

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-linux-arm64.tar.gz[Linux on ARM (aarch64, arm64)]

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-darwin-amd64.tar.gz[MacOS (x86_64, amd64)]

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-darwin-arm64.tar.gz[MacOS on ARM (aarch64, arm64)]

* link:https://mirror.openshift.com/pub/cgw/servicemesh/latest/istioctl-1.24.4-windows-amd64.zip[Windows (x86_64, amd64)]

. Extract the `istioctl` binary file.

> .. For the Linux operating system, wun the following command:

Reviewer 2: Typo

```suggestion
.. For the Linux operating system, run the following command:
```

Author: thanks for catching this!

done

> .. For the Linux operating system, run the following command:
> +
> [source,terminal]
> ----
> $ tar xzf istioctl-<VERSION>-<OS>-<ARCH>.tar.gz
> ----
> 
> .. For the Apple Mac Operating System, unpack and extract the archive.
> 
> .. For the Microsoft Windows Operating System, use the zip software to extract the archive.

Reviewer 3: Per [the ISG](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#indicating-optional-and-conditional-steps), conditional steps should be written one of two ways, one of the options being to start the step with an "if" statement. Could you rewrite these steps a little bit to better match that kind of structure?
```suggestion
.. If you are using an Linux operating system, run the following command:
+
[source,terminal]
----
$ tar xzf istioctl-<VERSION>-<OS>-<ARCH>.tar.gz
----

.. If you are using an Apple Mac Operating System, unpack and extract the archive.

.. If you are using an Microsoft Windows Operating System, use the zip software to extract the archive.
```

Feel free to slightly reword the steps, as long as they contain the "if" statement. (Or you can rewrite to use the other option that's more like "Condition: Step written as normal".)

Author: Done
